### DISCUSSION TITLE
Creation of server sockets with AF_UNIX SOCK_SEQPACKET type

### DISCUSSION THREAD

**Discussion by peter-rebyte** [kohlschutter/junixsocket#143] [General]
Hi, I've been trying to create a server socket that will listen and accept connections on the socket type as given in the title but have been running into some issues. I'm not sure this is supported today? It seems like maybe I could wrestle myself out of it by using raw fds but that seems a bit heavy handed. Is there something I'm missing?

**Comment by kohlschuetter** [comment]
SOCK_SEQPACKET behaves more like SOCK_DGRAM; it's packet-based but unlike DGRAM the order is guaranteed.

Can you try `AFUNIXDatagramSocket.newInstance(AFSocketType.SOCK_SEQPACKET)`, then bind (for the server) or connect (for the client)? If you want to use DatagramChannels, you can get them via `getChannel()`

**Comment by peter-rebyte** [comment]
Thans for the response, that's about where I was but I can't find a method that calls accept()?

**Comment by kohlschuetter** [comment]
Yeah, that looks like it's not implemented yet :-D
I will put something together, hang in there!

**Comment by kohlschuetter** [comment]
OK, please try latest from `main` (2.8.3-SNAPSHOT).

In commit 5d178b0670ee864a6eb2b2604662e984a1409414, I've added the necessary support for `listen`/`accept`.

See `junixsocket-common/src/test/java/org/newsclub/net/unix/domain/DatagramSocketTest.java` for an example how to use it. Please let me know how it goes!
